crates/bifrost/src/providers/replicated_loglet/tasks/digests.rs

```diff
@@ -325,14 +347,29 @@ impl Digests {
         );
         // walk backwards
         while let Some((offset, nodes)) = range.next_back() {
+            checker.fill_with_default();
             checker.set_attribute_on_each(nodes.iter().copied(), true);
             if checker.check_write_quorum(|known| *known) {
                 // this offset is good, advance to the next one
                 self.update_start_offset(offset.next());
                 return true;
             }
         }
-        false
+        // nothing to repair.
+        if self.is_finished() {
+            return true;
+        }
+        // when can we advance to repair from digest?
+        // - When we have digests enough to find at least a single copy of the oldest record in the
+        // repair range.
+        // - we have write-quorum of "writeable" nodes that responded to our digest requests (even if they
+        // don't have the records)
```

Comment:
Where is this condition checked?

Reply:
It looks that this happens outside of this method in the `RepairTail::run` method. That's why I was a bit confused.